File: app/src/main/java/com/zlm/kt/connect/ApiClient.java
package com.zlm.kt.connect;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class ApiClient extends BaseClient implements ApiClientImp {

    private static  ApiService api;

    @Inject
    public ApiClient() {
        api = createService(ApiService.class);
    }

    // -------------------------------------------
    @Override
    public ApiService getApi() {
        if (api == null) {
            synchronized (ApiService.class) {
                api = createService(ApiService.class);
            }
        }
        return api;
    }

}

File: app/src/main/java/com/zlm/kt/connect/ApiClient.kt
package com.zlm.kt.connect

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ApiClient @Inject constructor() : BaseClient(), ApiClientImp {
    // -------------------------------------------
    override val api: ApiService
        get() {
            if (Companion.api == null) {
                synchronized(ApiService::class.java) { Companion.api = createService(ApiService::class.java) }
            }
            return Companion.api
        }

    companion object {
        private lateinit var api: ApiService
    }

    init {
        Companion.api = createService(ApiService::class.java)
    }
}